src/lib/IFrameCommunicator.ts

```diff
@@ -68,20 +54,57 @@ export class IFrameCommunicator {
   public sendMessage = <M extends keyof IFrameMessagesMap>(
     type: M,
     data: Parameters<IFrameMessagesMap[M]>[0],
-    metadata?: Parameters<IFrameMessagesMap[M]>[1]
+    metadata?: Parameters<IFrameMessagesMap[M]>[1],
+    operationId?: string
   ): void => {
     if (!isInsideIFrame()) {
       console.error("Warning: can't send an iframe message, because the application is not hosted in an iframe");
       return;
     }
 
-    const message: IFrameMessage<M> = { type, data, metadata };
+    const message: IFrameMessage<M> = { type, data, metadata, operationId };
     window.parent.postMessage(message, '*');
   };
 
+  public sendMessageWithResponse = <M extends keyof IFrameMessagesMap>(
+    type: M,
+    data: Parameters<IFrameMessagesMap[M]>[0],
+    callback: () => void,
```

Comment:
Callback type should probably be taken from the IFrameMessagesMap for better type check as it is done for data and metadata (e.g. Parameters<IFrameMessagesMap[M]>[2]).